static/js/brand-store/Members/MembersTable.js

```diff
@@ -1,8 +1,61 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import PropTypes from "prop-types";
 import { MainTable, CheckboxInput } from "@canonical/react-components";
 
-function MembersTable({ filteredMembers }) {
+function MembersTable({ filteredMembers, changedMembers, setChangedMembers }) {
+  const [members, setMembers] = useState(filteredMembers);
+
+  const handleRoleChange = (member, role) => {
+    const updatedMembers = members.map((item) => {
+      let updatedItem = {};
+
+      if (item.id === member.id) {
+        if (item.roles.includes(role)) {
+          updatedItem = {
+            ...item,
```

Comment:
I'm not sure what the structure of `item` is but this is doing a shallow copy so you will want to be careful.

Reply:
I get your point. In this case `item` is a shallow object (is that a term 😉 ).

Reply:
Haha, I'm not sure, but now it is! You trendsetter you!